### PR TITLE
Update versions-compilers.rst

### PR DESCRIPTION
**Pull request by AlonaHlobina**
Added Java 16 (partially) and Python 3.9

### CODE REVIEW

File: docs/codeql/support/reusables/versions-compilers.rst [hunk 1]
```diff
@@ -17,11 +17,11 @@
 
    .NET 5","``.sln``, ``.csproj``, ``.cs``, ``.cshtml``, ``.xaml``"
    Go (aka Golang), "Go up to 1.16", "Go 1.11 or more recent", ``.go``
-   Java,"Java 7 to 15 [3]_","javac (OpenJDK and Oracle JDK),
+   Java,"Java 7 to 15 [3]_, Java 16 (partially)","javac (OpenJDK and Oracle JDK),
```
Comment (yo-h) [inline docs/codeql/support/reusables/versions-compilers.rst:20]:
```suggestion
   Java,"Java 7 to 16 [3]_","javac (OpenJDK and Oracle JDK),
```

Also update the footnote [3] further below to:
```
[3] Builds that execute on Java 7 to 16 can be analyzed. The analysis understands Java 15 standard language features.
```

**Review by yo-h — CHANGES_REQUESTED** [review]

File: docs/codeql/support/reusables/versions-compilers.rst (hunk 1 above)
Comment (yo-h) [reply docs/codeql/support/reusables/versions-compilers.rst]: Thanks for committing the suggested change. Please also update the footnote as above.

**Review by RasmusWL — APPROVED** [review]
:+1: Thanks, I had forgotten about this part of our docs :sweat_smile:

File: docs/codeql/support/reusables/versions-compilers.rst (hunk 1 above)
Comment (AlonaHlobina) [reply docs/codeql/support/reusables/versions-compilers.rst]: It should be fine now.

**Review by RasmusWL — APPROVED** [review]
still :+1: for Python

File: docs/codeql/support/reusables/versions-compilers.rst [hunk 2]
```diff
@@ -17,18 +17,18 @@
...
-   Python,"2.7, 3.5, 3.6, 3.7, 3.8",Not applicable,``.py``
+   Python,"2.7, 3.5, 3.6, 3.7, 3.8, 3.9",Not applicable,``.py``
    TypeScript [6]_,"2.6-4.2",Standard TypeScript compiler,"``.ts``, ``.tsx``"
 
 .. container:: footnote-group
 
     .. [1] Support for the clang-cl compiler is preliminary.
     .. [2] Support for the Arm Compiler (armcc) is preliminary.
-    .. [3] Builds that execute on Java 7 to 15 can be analyzed. The analysis understands Java 15 standard language features.
+    .. [3] Builds that execute on Java 7 to 16 can be analyzed. The analysis understands Java 16 standard language features.
```
Comment (yo-h) [inline docs/codeql/support/reusables/versions-compilers.rst:31]:
```suggestion
    .. [3] Builds that execute on Java 7 to 16 can be analyzed. The analysis understands Java 15 standard language features.
```
This difference is important. We do not yet fully support Java 16 language features, but we can extract builds that execute using the Java 16 compiler.

**Review by yo-h — CHANGES_REQUESTED** [review]

**Review by yo-h — APPROVED** [review]

**Comment by yo-h** [comment]
Merging following @RasmusWL's earlier approval and the additional change for Java.

**Pull request merged**